### PR TITLE
Go-modules, travis (AO-16731)

### PR DESCRIPTION
**Pull request by adamiklukasz**
https://swicloud.atlassian.net/browse/AO-16731

Summary of changes:
* Removed dep, added go.mod (v1, v2 and separate for examples)
* Fixed scripts ("cd PATH; go vet" instead of "go vet PATH") related to modules
* building v1/v2 examples

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -32,7 +32,7 @@ jobs:
       - TEST_TYPE=lint
     script:
       - make check 2>&1
-      - go install ./vendor/honnef.co/go/tools/cmd/staticcheck
+      - go get honnef.co/go/tools/cmd/staticcheck
```
Comment (lootek) [inline .travis.yml:35]: we use frozen version of staticcheck, please update

**Review by lootek — CHANGES_REQUESTED** [review]

File: scripts/common.sh [hunk 2]
```diff
@@ -104,7 +104,12 @@ _golint() {
 }
 
 _go_vet() {
-  go vet $(_test_dirs)
+  for d in $(_test_dirs)
+  do 
+    pushd $d
```
Comment (lootek) [inline scripts/common.sh:109]:
```suggestion
    pushd "$d"
```

File: scripts/common.sh [hunk 3]
```diff
@@ -104,7 +104,12 @@ _golint() {
 }
 
 _go_vet() {
-  go vet $(_test_dirs)
+  for d in $(_test_dirs)
```
Comment (lootek) [inline scripts/common.sh:107]:
```suggestion
  for d in "$(_test_dirs)"
```

File: scripts/common.sh (hunk 2 above)
Comment (lootek) [reply scripts/common.sh:109]: the same in other applicable places

File: scripts/deps.sh [hunk 4]
```diff
@@ -85,6 +86,9 @@ restore_go_dep() {
     dep)
       (cd "${__proj_dir}" && dep ensure -v -vendor-only)
       ;;
+    gomod)
+      (cd "${__proj_dir}" && go mod vendor)
```
Comment (lootek) [inline scripts/deps.sh:90]: please port that change to pluginsync (or make a JIRA issue)

File: v2/go.mod [hunk 5]
```diff
@@ -0,0 +1,17 @@
+module github.com/solarwinds/snap-plugin-lib/v2
+
+go 1.13
```
Comment (lootek) [inline v2/go.mod:3]: 1.15 is the current version

File: v2/go.mod [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+
+require (
+	github.com/StackExchange/wmi v0.0.0-20190523213315-cbe66965904d // indirect
```
Comment (lootek) [inline v2/go.mod:6]: what's that for in lib? @tomekkolo has just struggled to remove that from psutil/aosystem

File: go.mod [hunk 7]
```diff
@@ -0,0 +1,12 @@
+module github.com/solarwinds/snap-plugin-lib
+
+go 1.13
```
Comment (lootek) [inline go.mod:3]: 1.15

File: examples/v2/go.mod [hunk 8]
```diff
@@ -0,0 +1,9 @@
+module github.com/solarwinds/snap-plugin-lib/_examplesV2
+
+go 1.13
```
Comment (lootek) [inline examples/v2/go.mod:3]: 1.15

File: examples/v1/go.mod [hunk 9]
```diff
@@ -0,0 +1,10 @@
+module github.com/solarwinds/snap-plugin-lib/_examplesV1
+
+go 1.13
```
Comment (lootek) [inline examples/v1/go.mod:3]: 1.15

File: .travis.yml (hunk 1 above)
Comment (adamiklukasz) [reply .travis.yml]: done

File: scripts/common.sh (hunk 2 above)
Comment (adamiklukasz) [reply scripts/common.sh:109]: done

File: scripts/common.sh (hunk 3 above)
Comment (adamiklukasz) [reply scripts/common.sh:107]: done

File: v2/go.mod (hunk 6 above)
Comment (adamiklukasz) [reply v2/go.mod:6]: It's due to 
>> go mod why github.com/StackExchange/wmi
# github.com/StackExchange/wmi
github.com/solarwinds/snap-plugin-lib/v2/tutorial/07-proxy/collector/proxy
github.com/shirou/gopsutil/cpu
github.com/StackExchange/wmi

Maybe we should introduce a module for tutorial to widen dependencies

File: v2/go.mod (hunk 5 above)
Comment (adamiklukasz) [reply v2/go.mod:3]: but shouldn't we leave a minimal, compatible version here?

File: v2/go.mod (hunk 6 above)
Comment (adamiklukasz) [reply v2/go.mod]: https://github.com/solarwinds/snap-plugin-lib/pull/103/commits/f53f2af02b6c17aef1dcadc9fbfaa17c7f720352

File: scripts/deps.sh (hunk 4 above)
Comment (adamiklukasz) [reply scripts/deps.sh:90]: https://swicloud.atlassian.net/browse/AO-17506

File: scripts/common.sh (hunk 3 above)
Comment (adamiklukasz) [reply scripts/common.sh:107]: restored ... this one broke build

File: v2/go.mod (hunk 5 above)
Comment (lootek) [reply v2/go.mod:3]: nope. I see no reason for using such an old Go version

Comment (lootek) [reply v2/go.mod:3]: Go has a 1.x compatibility promise

File: examples/v1/go.mod (hunk 9 above)
Comment (lootek) [reply examples/v1/go.mod:3]: bump

**Review by lootek — CHANGES_REQUESTED** [review]

File: scripts/common.sh (hunk 3 above)
Comment (lootek) [reply scripts/common.sh:107]: hmm... wonder why? do you remember the error (just out of curiosity)

File: v2/go.mod (hunk 5 above)
Comment (lootek) [reply v2/go.mod:3]: discussed offline, agreed to add go versions to builder, resolving

**Review by lootek — DISMISSED** [review]

File: scripts/build.sh [hunk 10]
```diff
@@ -31,6 +31,9 @@ ubuntu_release=(lsb_release -rs)
 # shellcheck source=scripts/common.sh
 . "${__dir}/common.sh"
 
+# Build examples for v1 and v2
+for f in `find ./examples/ -name "main.go"`; do (cd ${f%\/*} && echo "Building $PWD" && go build) ; done
```
Comment (d-maslyk) [inline scripts/build.sh:35]: Please use rather `dirname` instead cutting parent directory from file name.
Also I would consider if "` `" for marking command substitution depreciated. (Please use rather $(foo)

Comment (adamiklukasz) [reply scripts/build.sh]: fixed

**Review by d-maslyk — APPROVED** [review]
Bash part for me looks OK.

**Review by tomekkolo — APPROVED** [review]

**Pull request merged**